packages/stylehacks/src/plugin.js

```diff
@@ -1,59 +1,59 @@
-export default function plugin(targets, nodeTypes, detect) {
-  class Plugin {
-    constructor(result) {
-      this.nodes = [];
-      this.result = result;
-      this.targets = targets;
-      this.nodeTypes = nodeTypes;
-    }
-
-    push(node, metadata) {
-      node._stylehacks = Object.assign({}, metadata, {
-        message: `Bad ${metadata.identifier}: ${metadata.hack}`,
-        browsers: this.targets,
-      });
+export default class BasePlugin {
+  constructor(targets, nodeTypes, result) {
```

Comment:
We need to place `result` last because sometimes plugins are constructed without passing one, so it could be undefined.